.github/workflows/dbld-images.yml

```diff
@@ -0,0 +1,65 @@
+name: Compile dbld-images
+
+on:
+  pull_request:
+  push:
+    paths:
+      - .github/workflows/dbld-images.yml
+      # List is from dbld/rules cache-image-% "WATCHED_FILES"
+      - "dbld/**"
+      - "packaging/rhel/syslog-ng.spec"
+      - "packaging/debian/control"
+  schedule:
+    - cron: '00 03 * * *'
+  workflow_dispatch:
+    inputs:
+      upload_images:
+        description: Should we upload the images into GitHub Packages? (true/false)
+        required: false
+        default: "false"
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    env:
+      CONTAINER_REGISTRY: "ghcr.io/${{ github.repository_owner }}"
+  
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Build the images
+        run: dbld/rules images image-devshell -j $(nproc)
+
+      - name: Should we upload the images?
+        run: |
+          if [[ "${{ github.event.inputs.upload_images }}" == "true" ]] || \
+             [ "${{ github.repository_owner }}" = "syslog-ng" ] && \
+             [ "${{ github.ref }}" = "refs/heads/master" ] && \
```

Comment:
This is the last time I comment on this line, I promise :)

It seems that || and && have the same operator precedence in bash, and they are left-associative, so this means:

Upload if upload is explicitly requested or push happened on the upstream master branch.